movie/src/main/java/com/calvinnor/movie/commons/util/NetworkUtils.kt

```diff
@@ -0,0 +1,52 @@
+package com.calvinnor.movie.commons.util
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+import android.os.Build
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+
+object NetworkUtils : ConnectivityManager.NetworkCallback() {
+
+    private val networkLiveData: MutableLiveData<Boolean> = MutableLiveData()
+
+    fun getNetworkLiveData(context: Context): LiveData<Boolean> {
```

Comment:
Looks like this function will be called multiple times, which will also register multiple listeners for Network callback?

I understand it's an object, so single instance, but please test this once. Rotate the phone, add a debugger to the callback and then trigger a network change. If only a single callback is received, we're good :)